engine/src/main/java/org/terasology/rendering/nui/widgets/ResettableUIText.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.rendering.nui.widgets;
+
+import org.terasology.math.geom.Rect2i;
+import org.terasology.rendering.nui.Canvas;
+import org.terasology.rendering.nui.SubRegion;
+import org.terasology.rendering.nui.UIWidget;
+
+public class ResettableUIText extends UIText {
+    private UIButton clearButton = new UIButton("clearButton", "X");
```

Comment:
This is a working approach, but IMO Terasology's default button doesn't look great when used as a small control element.

`UIDropdown` handles control elements slightly differently - it uses a texture ([dropdown.png](https://github.com/MovingBlocks/Terasology/blob/develop/engine/src/main/resources/assets/textures/ui/dropdown.png)) the right part of which masquerades as a button. I would use this idea for `ResettableUIText` as well:

- Create a modified version of [box.png](https://github.com/MovingBlocks/Terasology/blob/develop/engine/src/main/resources/assets/textures/ui/box.png) (the texture that `UIText` uses as a background) similar to dropdown.png. The left 80% of it should be the same as box.png, but the right 20% will contain a reset button. (Different percentage values are also fine - use what works best!)
- In `ResettableUIText.onDraw()` separate the canvas region into two parts:
  - The left (80%) will display the current text and place the cursor at a specified position when clicked on (same as the default `UIText`).
  - The right (20%) will reset the text field when clicked on. (This should also fix the issue mentioned in the PR as the "button" won't make the text field lose focus!)